lib/viewmodels/map/map_pin_view_model.dart

```diff
@@ -47,6 +47,15 @@ class MapPinViewModel extends AutoDisposeAsyncNotifier<List<MapPinDetails>> {
 
   /// Get nearby posts
   Future<List<MapPinDetails>> _getNearbyPosts() async {
+    //checking if the location services are enabled directly here so that
+    //we throw the same exception as the challenge case
+    final locationCheck =
+        await ref.read(geolocationServiceProvider).checkLocationServices();
+
+    if (locationCheck != null) {
+      throw locationCheck;
+    }
+
```

Comment:
Why do you do this, isn't this already done in the position stream provider ?

Reply:
Yes but the stream throws an exception first if i don't do it directly here. This results in different error messages when there is no location.

Reply:
okay that makes sense